packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts

```diff
@@ -569,10 +548,21 @@ export function createWorkerUploadForm(worker: CfWorkerInit): FormData {
 		keep_bindings.push(...keepBindings);
 	}
 
+	const assetConfig = {
+		html_handling: experimental_assets?.assetConfig?.html_handling,
+		not_found_handling: experimental_assets?.assetConfig?.not_found_handling,
+	};
+
+	const staticAssetOnly = experimental_assets?.routingConfig?.has_user_worker === false;
+	let mainModuleName = {}
+	if(main.type !== "commonjs" && !staticAssetOnly) {
+		mainModuleName = { main_module: main.name }
+	} else if (!staticAssetOnly) {
+		mainModuleName = { body_part: main.name }
+	}
+
 	const metadata: WorkerMetadata = {
-		...(main.type !== "commonjs"
-			? { main_module: main.name }
-			: { body_part: main.name }),
+		...mainModuleName,
 		bindings: metadataBindings,
```

Comment:
Someone could add bindings etc. to a static-asset only project and this probably would not error. It also wouldn't have any affect because EWC won't save anything besides compat date/flags.
Is this something we want to warn users about?

Reply:
already being tracked here for after BW https://jira.cfdata.org/browse/WC-2562